src/main/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtil.java

```diff
@@ -30,23 +32,25 @@
 
 public class AutoSetFileLinksUtil {
 
+    record RelinkedResults(List<LinkedFile> relinkedFiles, List<String> exceptions) { }
+
     public static class LinkFilesResult {
         private final List<BibEntry> changedEntries = new ArrayList<>();
-        private final List<IOException> fileExceptions = new ArrayList<>();
+        private final List<String> fileExceptions = new ArrayList<String>();
 
         protected void addBibEntry(BibEntry bibEntry) {
             changedEntries.add(bibEntry);
         }
 
-        protected void addFileException(IOException exception) {
+        protected void addFileException(String exception) {
             fileExceptions.add(exception);
         }
 
         public List<BibEntry> getChangedEntries() {
             return changedEntries;
         }
 
-        public List<IOException> getFileExceptions() {
+        public List<String> getFileExceptions() {
```

Comment:
Why reduction to weaker type? `String` should not be used if more finer-grainded data types are available.

See https://www.pearson.de/effective-java-9780134686073